### ISSUE TITLE
Unable to translate Ruby 2.6

### ISSUE THREAD

**Issue by thedataking** [immunant/c2rust#78]
Reproducing on Ubuntu: 
     
    apt-get install bison ruby
    curl https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.3.tar.gz | tar zx && ruby-2.6.3/
    ./configure --disable-shared --disable-install-doc --disable-install-rdoc \
         --disable-install-capi --disable-jit-support -with-out-ext=dbm warnflags=-w
    intercept-build make -j`ncpus` miniruby
    c2rust transpile -m ruby compile_commands.json

Error:
```
Transpiling st.c
error: Skipping declaration Some(Struct { name: Some("st_hash_type"), fields: Some([CDeclId(8843), CDeclId(8844)]), is_packed: false, manual_alignment: None, max_field_alignment: None, platform_byte_size: 16, platform_alignment: 8 }) due to error: Unable to convert function pointer type without prototype

thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:345:21
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:381
   6: rust_begin_unwind
             at src/libstd/panicking.rs:308
   7: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
   8: core::panicking::panic
             at src/libcore/panicking.rs:49
   9: c2rust_transpile::translator::literals::<impl c2rust_transpile::translator::Translation>::convert_init_list
  10: c2rust_transpile::translator::Translation::convert_expr
  11: c2rust_transpile::translator::Translation::convert_variable
  12: c2rust_transpile::translator::Translation::convert_decl
  13: scoped_tls::ScopedKey<T>::set
  14: syntax::with_globals
  15: c2rust_transpile::translator::translate
  16: c2rust_transpile::transpile_single
  17: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
  18: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter
  19: c2rust_transpile::transpile
  20: c2rust_transpile::main
  21: std::rt::lang_start::{{closure}}
  22: std::panicking::try::do_call
             at src/libstd/rt.rs:49
             at src/libstd/panicking.rs:293
  23: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:87
  24: std::rt::lang_start_internal
             at src/libstd/panicking.rs:272
             at src/libstd/panic.rs:388
             at src/libstd/rt.rs:48
  25: main
  26: __libc_start_main
  27: _start
````

Problematic code (st.c:142-146):
```C
 #define type_numhash st_hashtype_num
 static const struct st_hash_type st_hashtype_num = {
     st_numcmp,
     st_numhash,
 };
```

Clang AST:
<img width="1226" alt="clang-ast-dump-for-st" src="https://user-images.githubusercontent.com/4011337/56464076-77f06c00-6396-11e9-9bc9-1acb497bbe35.png">

**Comment by thedataking** [comment]
Blocked on #85.

**Comment by rinon** [comment]
Fixed by 23c1859e67e0b3091dbbae06e0169a52cd18f7e5
